src/directional_clustering/plotters/lines.py

```diff
@@ -0,0 +1,120 @@
+from compas.geometry import normalize_vector
+from compas.geometry import add_vectors
+from compas.geometry import scale_vector
+from compas.geometry import length_vector
+
+
+__all__ = [
+    "line_sdl",
+    "vector_lines_on_faces",
+    "line_tuple_to_dict",
+    "polygon_list_to_dict"
+]
+
+
+def line_sdl(start, direction, length, both_sides=True):
+    """
+    Creates a line from a start point with a given direction and length. It will extend the line in the oposite direction as well unless both_sides is set to False.
+
+    Input
+    -----
+    start : `array`, shape(n,), n<=3
+        A point defined by XYZ coordinates.
+        If Z coordinate is not given, then the results are in 2D.
+        If Y and Z coordinates are not given, then the results are in 1D.
+    direction : `array`
```

Comment:
If we use `array`, in Python I think it has a `numpy` connotation which we aren't using here. What about we simplify and use `list` of `float` instead?